assets/js/blocks/legacy-template/utils.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { TemplateAttributes } from './types';
+
+export function beginsWith( needle: string, haystack: string ): boolean {
+	return haystack.substr( 0, needle.length ) === needle;
+}
```

Comment:
🤔 Why not using [`.startsWith()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith)?

Reply:
I started off doing, then considered compatibility with IE, and then remembered we dropped support for IE so I could have used it after all. I will replace, and if the unit tests for `getMatchingTemplateData` still pass I'll merge. Thanks!